src/resources/views/crud/buttons/delete.blade.php

```diff
@@ -50,16 +35,8 @@ className: "bg-danger",
 		                    text: "{!! '<strong>'.trans('backpack::crud.delete_confirmation_title').'</strong><br>'.trans('backpack::crud.delete_confirmation_message') !!}"
 		                  }).show();
 
-			              // Hide the modal, if any
-			              $('.modal').modal('hide');
-
-			              // Remove the details row, if it is open
-			              if (row.hasClass("shown")) {
-			                  row.next().remove();
-			              }
-
 			              // Remove the row from the datatable
-			              row.remove();
+						  crud.table.row($("#crudTable a[data-route='"+route+"']").parents('tr')).remove().draw();
```

Comment:
Hello @martijnb92 

Thanks again for one more contribution. I was testing this out, and I am afraid that this is a change we cannot aford to do. 

Redrawing the table is a very expensive operation, if you want to have a better understanding of what I mean load 200 entries in the table and delete one, using both versions of code. (plus the cost of drawing the table goes up as the number of columns go up too not only the number of entries).

Let me know @martijnb92 :)

Reply:
I did some testing with a table with 350 entries and 10 columns, never did it take more than a second to redraw the table, even on my phone.
The old version of the code is a bit faster/instant, since it only removes the table row. But that means the totale entries line at the top is incorrect, since it’s not updated on delete.
That’s why in my opinion the re-draw option in datatable is a good idea, since it actually removes the data instead of just removing the rendered table row.

Reply:
Thanks for clarification on this. 

Unfortunatelly we are aware of some issues in Datatables that make some tables take some amount of time to load. It's a combination of the number of columns, type of columns, number of entries etc, so I must sustain what I said before:

- the redraw option is only good in already fast tables, if you happen to have a table that takes 4 seconds to load, it will take 4 seconds to redraw (4 seconds it's not an HUGE issue in real big data cruds), but it will be 4 times faster to do it like previously than redrawing the table. 

Let's keep in mind what you mentioned: ` the totale entries line at the top is incorrect, since it’s not updated on delete`, are you aware of someway we could update this value upon deletion or is redrawing the only option ? 

Thanks for your effort in solving this, I am pretty sure we will end this with a better solution than what we have in current backpack code 👍

Reply:
Hmmm good thinking... both of you 😂
- re-drawing the table is indeed more expensive;
- that number _is_ wrong, so... this is a bug; a small one, but a bug nonetheless;

Tricky one...

Reply:
Ok so... here's how I see it:
- Personally I don't see how we'd be able to redraw just that number, instead of redrawing the whole table. Not without breaking changes which are definitely not worth it.
- I don't think it's worth trying to just change the number at the top, because in some cases, the `count()` operation is the most expensive one (think tables with millions upon millions of entries - over there getting 10 entries is fast, counting 99 million entries is what's expensive). 
- Decrementing that number with JS... yeah... that feels like a hack so I'd rather we not do that either.
- Redrawing the datatable also prevents this one situation, that was annoying to me in the past: deleting a bunch of stuff one by one leads you to look at an empty table;
- Redrawing the datatable also fixes an inconsistency we currently have: 
    - Delete doesn't do a table redraw
    - BulkDelete does do a datatable redraw; 
(so if you use both Delete and BulkDelete, the not-redrawing thing looks like a bug, not a feature)

To me, it looks like the redraw is more expensive indeed, but it's a reasonable drawback for fixing this bug (and a few other minor bugs in the process). 

I think the ones considerably affected by the redraw will be CRUDs paginated by 100+ items per page and/or with a large number of columns. But if redrawing is a problem there, it means it was also a problem at the first page load. And upon search. And upon filtering. And upon reordering. So it's a problem worth fixing.

I say... let's redraw 🥳 I'm still open to being convinced otherwise @pxpm if you feel strongly about this.

Reply:
That being said, I did find a bug with redrawing the table. And I've noticed it's also a bug inside the BulkDelete operation, so we could fix it there too.

---

If you're on page 33/100, and you delete an item, you'd expect the item to be deleted but stay on page 33/100. But no, the way this PR redraws the table, the user is sent to page 1. Which I think will make it super-frustrating to users.

![2020-12-09 11 34 10](https://user-images.githubusercontent.com/1032474/101658816-c47d3500-3a1b-11eb-8c70-fe81c73a0deb.gif)

I've checked filters, they are kept.
I've checked search, that is kept.
But the page number is not, that is reset upon datatable redraw.